torch/csrc/autograd/functions/tensor.cpp

```diff
@@ -30,14 +30,8 @@ auto CopyBackwards::apply(variable_list&& grads) -> variable_list {
         grad = c10::MaybeOwned<at::Tensor>::owned(at::real(grads[0]));
       }
 
-      at::DeviceGuard device_guard(src_device);
-      // TODO: What if !grad.is_cuda(), but src_device is CUDA?
-      // This code is kind of weirdly asymmetric.
-      const bool copy = (grad->is_cuda() && grad->device() != src_device);
-      grad_inputs[1] = grad->to(
-          src_options,
-          /*non_blocking=*/false,
-          /*copy=*/copy);
+      at::DeviceGuard device_guard(src_options.device());
```

Comment:
I'm also not sure this `DeviceGuard` is necessary.

Reply:
I think it is. The backward ops is properly guarded for the first output's device. But the input's device might need to get updated here (for stream in particular)